carbonfive/raygun-rails
# Upgrade Rails 6.1 → 7.0 and Ruby 3.0 → 3.1

Problem
=======

Rails 7 and Ruby 3.1 have been out for several months, and Rails ecosystem of dependencies seems to have stabilized. Time to upgrade!

Solution
========

1. Run `bundle update` to get the latest Rails 6.1 compatible gems.
1. Change the Gemfile pin from `"rails", "~> 6.1"` to `"rails", "~> 7.0.3"` and run `bundle update`.
1. Remove the `listen` gem, since that is no longer recommended with Rails 7.
1. Run `bin/rails app:update` and adopt the suggestions that make sense.
1. Enable the new Rails 7 framework defaults by setting `config.load_defaults 7.0`.
1. Run `bin/rails db:migrate` to regenerate `db/schema.rb`.
1. Run `npx browserslist@latest --update-db` to fix caniuse-lite warnings.
1. Upgrade to Ruby 3.1.2

Steps to Verify
---------------

1. Run `bin/setup`
1. Run `bin/rake` and confirm tests pass
1. Run `yarn start` and confirm the app is accessible on <http://localhost:3000>

I also deployed to Heroku and confirmed it works out of the box.
